src/options/smt_options.toml

```diff
@@ -254,20 +254,12 @@ name   = "SMT Layer"
   default    = "false"
   help       = "support the get-assignment command"
 
-[[option]]
-  name       = "interactiveMode"
-  long       = "interactive-mode"
-  category   = "undocumented"
-  type       = "bool"
-  predicates = ["setProduceAssertions"]
-  help       = "deprecated name for produce-assertions"
-
 [[option]]
   name       = "produceAssertions"
   category   = "common"
   long       = "produce-assertions"
   type       = "bool"
-  predicates = ["setProduceAssertions"]
+  alias      = ["interactive-mode"]
```

Comment:
What exactly is this doing?  Is this saying that produce assertions is true when interactive-mode is true?  Can this be refactored not to rely on the toml?

Reply:
This makes `--produce-assertions` and `--interactive-mode` are different names for the same thing. This actually preserves current behaviour, as they have, in fact, always the same value; this is, however, achieved in a weird and very convoluted way right now...

Reply:
Really? I would perhaps expect that produce-assertions is *implied* by interactive-mode but not equivalent to it?

Regardless this is indeed very convoluted.

Reply:
This handler was called by both options: https://github.com/cvc5/cvc5/pull/7295/files#diff-831a3e7302809583cb776be88353361408845885dc7d393ca45cad5e069d7e18L509 ...

Reply:
What do you want it to do? Have `--interactive-mode` imply `--produce-assertions`?
Note that the the value behind `--interactive-mode` is not used in all of cvc5, from all that I can tell...

Reply:
If its not used, then can we delete `interactive-mode`?

Otherwise maybe it'd make sense to have it imply produce-assertions but not vice versa.

Reply:
The current change was my defensive way of essentially deleting `--interactive-mode`, keeping it around as an alias in case someone uses it. Fwiw, you can simply use `--produce-assertions` to do the very same thing.

Reply:
I think I'd outright delete it if no one is using it.